core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -141,6 +144,13 @@ public static ValueReader<byte[]> decimalBytesReader(Schema schema) {
     }
   }
 
+  @SuppressWarnings("unchecked")
+  public static ValueReader<Variant> variants(
+      ValueReader<?> metadataReader, ValueReader<?> valueReader) {
+    return new VariantReader(
```

Comment:
Similarly, can we reuse the same VariantReader instance?

Reply:
Fixed.